### PR TITLE
Fix Cutout2D with CompImageHDU.section

### PR DESCRIPTION
**Pull request by astrofrog**
Fixes https://github.com/astropy/astropy/issues/14997

### CODE REVIEW

File: astropy/nddata/utils.py [hunk 1]
```diff
@@ -592,7 +592,9 @@ def __init__(
                         "shape can contain Quantities with only pixel or angular units"
                     )
 
-        if not isinstance(data, (Section, CompImageSection)):  # Accept lazy-loaded image sections
+        if not isinstance(
+            data, (Section, CompImageSection)
```
Comment (pllim) [inline astropy/nddata/utils.py:596]: Where is the fix? I only see style changes.

Comment (pllim) [reply astropy/nddata/utils.py:596]: Oh nvm, GitHub took me to only one of the commits diff. Ignore me...

**Comment by pllim** [comment]
I think no need to backport to LTS? If you disagree, please change the milestone and update backport labels. Thanks!

**Comment by astrofrog** [comment]
Yes no need to backport as there is no .section for compressed images in LTS

**Comment by astrofrog** [comment]
pre-commit.ci autofix

File: astropy/nddata/tests/test_utils.py [hunk 2]
```diff
@@ -605,3 +606,32 @@ def test_cutout_with_nddata_as_input(self):
...
+    # Make sure that one can pass ImageHDU.section and CompImageHDU.section
+    # to Cutout2D
+
+    data = np.ones((200, 200))
+
+    hdu = fits.ImageHDU(data=data)
+    hdu.writeto(tmp_path / "uncompressed.fits")
+
+    with fits.open(tmp_path / "uncompressed.fits") as hdul:
+        header = hdul[1].header
```
Comment (saimn) [inline astropy/nddata/tests/test_utils.py:621]: No need to access `.header` here ? 
```suggestion
```

File: astropy/nddata/tests/test_utils.py [hunk 3]
```diff
@@ -605,3 +606,32 @@ def test_cutout_with_nddata_as_input(self):
...
+            hdul[1].section,
+            (75, 75),
+            100 * u.pix,
+        )
+
+    chdu = fits.CompImageHDU(data=data)
+    chdu.writeto(tmp_path / "compressed.fits")
+
+    with fits.open(tmp_path / "compressed.fits") as hdul:
+        header = hdul[1].header
```
Comment (saimn) [inline astropy/nddata/tests/test_utils.py:632]:
```suggestion
```

**Review by saimn — COMMENTED** [review]
Thqnks for the fix @astrofrog, just a minor comment.

**Review by saimn — APPROVED** [review]

**Pull request merged**